azure/azure-endpoints-availability/azure_connectivity_checker.sh

```diff
@@ -0,0 +1,198 @@
+#!/bin/bash
+
+# Azure endpoints to check
+endpoints=("https://management.azure.com/" "https://login.microsoftonline.com/" "https://management.core.windows.net/")
+isProxyUsed=false
+config_dir=""
+ssl_cert=""
+silent_mode=false
+
+# Create a folder named "azure_connectivity_logs" if it doesn't exist
+log_folder="/var/lib/dynatrace/gateway/temp/azure_connectivity_logs" || { echo "Error creating log folder."; exit 1; }
+mkdir -p "$log_folder" || { echo "Error creating log folder for user [$USER]"; exit 1; }
+
+# Output file with timestamp will be store in the log folder
+timestamp=$(date +"%Y%m%d_%H%M%S")
+output_file="${log_folder}/azure_connectivity_check_${timestamp}.log"
+touch "$output_file" || { echo "Error creating output file for user [$USER]"; exit 1; }
+
+# Log to console and file
+log_to_console_and_file() {
+    echo -e "$1" | tee -a "$output_file"
+}
+
+# Log to file only
+log_to_file() {
+    echo -e "$1" >> "$output_file"
+}
+
+log_message() {
+    local message="$1"
+    if [ "$silent_mode" = true ]; then
+        log_to_file "$message"
+    else
+        log_to_console_and_file "$message"
+    fi
+}
+
+log_separator="-----------------------------------------------------------------------"
+log_separator2="======================================================================="
+
+log_to_console_and_file "$log_separator2"
+log_to_console_and_file "Azure Connectivity Checker Script"
+log_to_console_and_file "$log_separator2"
+# Detect current working directory and user
+log_to_console_and_file "Working directory: $PWD"
+log_to_console_and_file "Current user: $USER"
+
+# Read script parameters or set default values to directories
+OPTSTRING=":p:c:s"
+
+while getopts ${OPTSTRING} opt; do
+    case ${opt} in
+    p)
+       config_dir=${OPTARG}
+       ;;
+    c)
+        ssl_cert=${OPTARG}
+        ;;
+    s)
+        silent_mode=true
+        ;;
+    :)
+        echo "Option -${OPTARG} requires an argument."
+        exit 1
+        ;;
+    ?)
+        echo "Invalid option -${OPTARG}"
+        exit 1
+        ;;
+    esac
+done
+
+direct_connect_command() {
+  local endpoint="$1"
+  curl_command="curl -v -I $endpoint --max-time 5"
+  echo $curl_command
+}
+
+proxy_connect_command() {
+  local endpoint="$1"
+  local proxy_server="$2"
+  local proxy_port="$3"
+  local cert="$4"
+
+  curl_command="curl -v -x https://$proxy_server:$proxy_port $endpoint:443 --proxy-cacert $cert"
+  echo $curl_command
+}
+
+check_connectivity() {
+    log_to_console_and_file "Checking the availability of Azure endpoints"
+    log_to_console_and_file "$log_separator"
+
+    for endpoint in "${endpoints[@]}"; do
+        if [ "$isProxyUsed" = true ]; then
+            # Check connection to Azure via proxy
+            log_to_console_and_file "Checking connectivity to Azure endpoint [$endpoint:443] via [$proxy_server:$proxy_port]..."
+            curl=`proxy_connect_command "$endpoint" "$proxy_server" "$proxy_port" "$ssl_cert"`
```

Comment:
is the script overriding the `curl` command?

Reply:
It was supposed to be a variable called `curl.`  the method `proxy_connect_command` returns a prepared string for the command I want to use later (with or without proxy settings).
I am not a big expert in bash, so if it overrides the command, it can be an issue. But the script worked with this `curl` as a variable... It might be worth changing this variable's name just to be sure.

Reply:
it won't override the command, as the variable needs to be accessed with $, but anyway I think it's worth doing the name change